### PR TITLE
feat: rule creation guidelines for sigmahq

### PR DESCRIPTION
**Pull request by nasbench**
Since we enforce certain fields and conditions on the rules hosted on SigmaHQ. We need another version of the specs stating the enforced changes.

This is the first attempt to provide such a file.

The aim is to be able to link to this file when discussing certain aspects of rule creation during PRs and such.

This will also be linked via the PR template in SigmaHQ

### CODE REVIEW

File: sigmahq/sigmahq_specification.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+## Status
+
+All newly created rules must start with a status of `experimental`
+
+## References
+
+All rules must provide a public reference when possible
+
+## Single Item Lists
```
Comment (frack113) [inline sigmahq/sigmahq_specification.md:72]:
```suggestion
## Detection

### Item Lists
```

File: sigmahq/sigmahq_specification.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+Example of multi item list:
+
+```yml
+detection:
+    selection:
+        Image|endswith:
+            - '\example_1.exe'
+            - '\example_2.exe'
+            - '\example_3.exe'
+```
```
Comment (frack113) [inline sigmahq/sigmahq_specification.md:93]:
```suggestion
`

### Selection name

to keep it easy to read, use for the simple rules the name 'selection' and 'filter'
`yml
selection:
filter:
condition: selection and not filter
`
```

Can not put the 3 `

File: sigmahq/sigmahq_specification.md [hunk 3]
```diff
@@ -0,0 +1,95 @@
+# SigmaHQ Custom Specification <!-- omit in toc -->
+
+This document describe an addtional set of custom requirements enfroced by the SigmaHQ rule repository in order to ensure an easy to maintain rule base. For the general Sigma specficiation please read the [Sigma_specification.md](../Sigma_specification.md).
```
Comment (phantinuss) [inline sigmahq/sigmahq_specification.md:3]:
```suggestion
This document describes an additional set of custom requirements enforced by the SigmaHQ rule repository in order to ensure an easy to maintain rule base. For the general Sigma specification please read the [Sigma_specification.md](../Sigma_specification.md).
```

File: sigmahq/sigmahq_specification.md [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+level [required]
+```
+
+## Filenames
+
+All rule filename must follow the convention described in [Sigmahq_filename_rule.md](./Sigmahq_filename_rule.md)
+
+## Titles
+
+All SigmaHQ rules must titles must use title casing
```
Comment (phantinuss) [inline sigmahq/sigmahq_specification.md:56]:
```suggestion
All SigmaHQ rule titles must use title casing
```

**Comment by phantinuss** [comment]
I would argue to not see this as part of the specification. It is just a guideline / convention of the rules repo. We should be able to change it at any time without any impact for future or previous versions of the spec. I see it like "coding conventions" or "coding styles" that are used in bigger projects.

File: sigmahq/sigmahq_specification.md [hunk 5]
```diff
@@ -0,0 +1,95 @@
+# SigmaHQ Custom Specification <!-- omit in toc -->
```
Comment (phantinuss) [inline sigmahq/sigmahq_specification.md:1]:
```suggestion
# SigmaHQ Rule Conventions <!-- omit in toc -->
```

File: sigmahq/sigmahq_specification.md [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+
+This document describes an additional set of custom requirements enforced by the SigmaHQ rule repository in order to ensure an easy to maintain rule base. For the general Sigma specification please read the [Sigma_specification.md](../Sigma_specification.md).
```
Comment (phantinuss) [inline sigmahq/sigmahq_specification.md:3]:
```suggestion
This document describes an additional set of rule conventions enforced by the SigmaHQ rule repository in order to ensure an easy to maintain rule base. For the general Sigma specification please read the [Sigma_specification.md](../Sigma_specification.md).
```

**Review by phantinuss — COMMENTED** [review]
As a suggestion

File: sigmahq/sigmahq_specification.md (hunk 6 above)
Comment (phantinuss) [reply sigmahq/sigmahq_specification.md:3]: If accepted, the file name should change as well

**Comment by nasbench** [comment]
> I would argue to not see this as part of the specification. It is just a guideline / convention of the rules repo. We should be able to change it at any time without any impact for future or previous versions of the spec. I see it like "coding conventions" or "coding styles" that are used in bigger projects.

There is nothing preventing us from changing this in the future. It's hosted here in a folder called `sigmahq` for this reason. You could call it a convention but since it's enforced via test it cannot be a convention anymore. 

TBH, I don't really care about the title as long as we link to it in the PR template so that people know in advance how they should write.

**Comment by phantinuss** [comment]
Conventions can be enforced. If I want to add to the Linux kernel and don't use the right number of spaces for indentation, the PR will not be accepted.

We agree in it being not part of the Sigma spec. So not calling it a spec at all helps in either way to not confuse it.

**Review by phantinuss — APPROVED** [review]

**Review by frack113 — APPROVED** [review]

**Review by Neo23x0 — APPROVED** [review]

**Pull request merged**